test/widgets/message_list_test.dart

```diff
@@ -1755,6 +1775,64 @@ void main() {
       debugNetworkImageHttpClientProvider = null;
     });
 
+    group('User status emoji', () {
+      void checkStatusEmoji(User user, Type type, {required bool isPresent}) {
+        final nameFinder = find.text(user.fullName);
+        final statusEmojiFinder = findStatusEmoji(type);
+
+        check(nameFinder).findsOne();
+        check(statusEmojiFinder).findsExactly(isPresent ? 1 : 0);
+        if (isPresent) {
+          checkUserStatusEmoji(statusEmojiFinder, isAnimated: false);
+        }
+
+        final senderRowFinder = find.ancestor(of: nameFinder,
+          matching: find.ancestor(of: statusEmojiFinder,
+            matching: find.byType(Row)));
+        isPresent
+          ? check(senderRowFinder).findsAny()
+          : check(senderRowFinder).findsNothing();
```

Comment:
Instead of these three separate if-isPresent conditions, I think it'd be clearer to have one if/else on isPresent.

In particular, when isPresent is false, only the first check does anything. So with an if/else, the rest can just be left out in that branch.